Dockerfile

```diff
@@ -13,49 +13,19 @@ FROM ubuntu:focal as base
 # Install system requirements.
 # We update, upgrade, and delete lists all in one layer
 # in order to reduce total image size.
-RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --yes \
-        # Global requirements
-        build-essential \
-        curl \
-        # If we don't need gcc, we should remove it.
-        g++ \
-        gcc \
-        git \
-        git-core \
-        language-pack-en \
-        libfreetype6-dev \
-        libmysqlclient-dev \
-        libssl-dev \
-        libxml2-dev \
-        libxmlsec1-dev \
-        libxslt1-dev \
-        swig \
-        # openedx requirements
-        gettext \
-        gfortran \
-        graphviz \
-        libffi-dev \
-        libfreetype6-dev \
-        libgeos-dev \
-        libgraphviz-dev \
-        libjpeg8-dev \
-        liblapack-dev \
-        libpng-dev \
-        libsqlite3-dev \
-        libxml2-dev \
-        libxmlsec1-dev \
-        libxslt1-dev \
-        # lynx: Required by https://github.com/openedx/edx-platform/blob/b489a4ecb122/openedx/core/lib/html_to_text.py#L16
-        lynx \
-        ntp \
-        pkg-config \
-        python3-dev \
-        python3-venv && \
-    rm -rf /var/lib/apt/lists/*
-
-# Set locale.
-RUN locale-gen en_US.UTF-8
+ARG APT_MANDATORY_REQUIREMENTS="python3 python3.8 python3.8-minimal python3-venv libpython3.8 libpython3.8-stdlib libmysqlclient21 libssl1.1 libxmlsec1-openssl lynx ntp gettext gfortran graphviz locales swig"
+ARG APT_BUILD_REQUIREMENTS="build-essential curl git git-core pkg-config libmysqlclient-dev libssl-dev libxml2-dev libxmlsec1-dev libxslt1-dev python3-dev \
```

Comment:
I'm not sure if I like these squashed to fewer lines, do we know why we need the -dev package and if we can use a non-dev version of any of them?  cc @jmbowman or @robrap